### PR TITLE
Hoverable Preview component

### PR DESCRIPTION
**Pull request by Arcyvilk**
Related ticket: [click!](https://cognitedata.atlassian.net/browse/CDF-3457)

Generic component for the hoverable preview of assets etc., still WIP

### CODE REVIEW

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-640.gearboxjs.preview.cogniteapp.com

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-640.gearboxjs.preview.cogniteapp.com

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-640.gearboxjs.preview.cogniteapp.com

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-640.gearboxjs.preview.cogniteapp.com

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-640.gearboxjs.preview.cogniteapp.com

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-640.gearboxjs.preview.cogniteapp.com

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-640.gearboxjs.preview.cogniteapp.com

**Review by ghost — APPROVED** [review]
Pretty 🤩 
please wait for someone on gearbox team to provide feedback

File: src/components/HoverablePreview/types.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+
+type TBorderType = 'top' | 'right' | 'bottom' | 'left';
+export type TBorders = TBorderType[];
+export type TAlign = 'left' | 'right' | 'center';
```
Comment (dlcognite) [inline src/components/HoverablePreview/types.ts:5]: wow you thought of everything 🚀

**Comment by mrtraser** [comment]
Hey, @agadacz-cognite , could you take a look at [Popover](https://ant.design/components/popover/) from Ant Design? Looks like it can be used instead of `HoverablePreview`, doesn't it?

**Comment by agadacz-cognite** [comment]
> Hey, @agadacz-cognite , could you take a look at [Popover](https://ant.design/components/popover/) from Ant Design? Looks like it can be used instead of `HoverablePreview`, doesn't it?

That's a good call, it removes a bit of repeated logic.

**Comment by dlcognite** [comment]
I think eventually we want to move away from AntD.

**Comment by mrtraser** [comment]
@dlcognite, If you move from AntD then you'll not able to use gearbox, right? Cause AntD is a peer dependency for gearbox

**Comment by mrtraser** [comment]
> That's a good call, it removes a bit of repeated logic.

Probably you can just replace the whole `HoverablePreview` component with the component from AntD. [Popover](https://ant.design/components/popover/#API) allows you to provide react component as the content of popover

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-640.gearboxjs.preview.cogniteapp.com

**Comment by hhenriques** [comment]
Hi @agadacz-cognite, I spoke with David this fits better in Cogs. I just wanted to say that the component looks pretty good and I especially liked the documentation!

**Comment by agadacz-cognite** [comment]
> Hi @agadacz-cognite, I spoke with David this fits better in Cogs. I just wanted to say that the component looks pretty good and I especially liked the documentation!

Thanks <3
I am closing this PR too - the entire functionality is moved to [Cogs.js](https://github.com/cognitedata/cogs.js/pull/136).

**Comment by polomani** [comment]
@agadacz-cognite you did a great job btw :) We just need to communicate better next time)

**Pull request closed without merging**